test/integration/targets/sts_assume_role/tasks/main.yml

```diff
@@ -0,0 +1,382 @@
+
+    - name: assert with no parameters
+      assert:
+        that:
+           - 'result.failed'
+           - "'missing required arguments:' in result.msg"
+
+    # ============================================================
+    - name: test with empty parameters
+      sts_assume_role:
+        aws_access_key: "{{ aws_access_key }}"
+        aws_secret_key: "{{ aws_secret_key }}"
+        security_token: "{{ security_token }}"
+        region: "{{ aws_region}}"
+        role_arn:
+        role_session_name:
+        policy:
+        duration_seconds:
+        external_id:
+        mfa_token:
+        mfa_serial_number:
+      register: result
+      ignore_errors: true
+
+    - name: assert with empty parameters
+      assert:
+        that:
+           - 'result.failed'
+           - "'Missing required parameter in input:' in result.msg"
+      when: result.module_stderr is not defined
+
+    - name: assert with empty parameters
+      assert:
+        that:
+          - 'result.failed'
+          - "'Member must have length greater than or equal to 20' in result.module_stderr"
+      when: result.module_stderr is defined
+
+    # ============================================================
+    - name: test with only 'role_arn' parameter
+      sts_assume_role:
+        aws_access_key: "{{ aws_access_key }}"
+        aws_secret_key: "{{ aws_secret_key }}"
+        security_token: "{{ security_token }}"
+        role_arn: "{{ test_role.iam_role.arn }}"
+      register: result
+      ignore_errors: true
+
+    - name: assert with only 'role_arn' parameter
+      assert:
+        that:
+           - 'result.failed'
+           - "'missing required arguments: role_session_name' in result.msg"
+
+    # ============================================================
+    - name: test with only 'role_session_name' parameter
+      sts_assume_role:
+        aws_access_key: "{{ aws_access_key }}"
+        aws_secret_key: "{{ aws_secret_key }}"
+        security_token: "{{ security_token }}"
+        role_session_name: "AnsibleTest"
+      register: result
+      ignore_errors: true
+
+    - name: assert with only 'role_session_name' parameter
+      assert:
+        that:
+           - 'result.failed'
+           - "'missing required arguments: role_arn' in result.msg"
+
+    # ============================================================
+    - name: test assume role with invalid policy
+      sts_assume_role:
+        aws_access_key: "{{ aws_access_key }}"
+        aws_secret_key: "{{ aws_secret_key }}"
+        security_token: "{{ security_token }}"
+        region: "{{ aws_region}}"
+        role_arn: "{{ test_role.iam_role.arn }}"
+        role_session_name: "AnsibleTest"
+        policy: "invalid policy"
+      register: result
+      ignore_errors: true
+
+    - name: assert assume role with invalid policy
+      assert:
+        that:
+          - 'result.failed'
+          - "'Not authorized to perform sts:AssumeRole' in result.msg"
```

Comment:
This fails for me at this point:

```
TASK [sts_assume_role : test assume role with invalid policy] ******************
task path: /root/ansible/test/integration/targets/sts_assume_role/tasks/main.yml:119
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: MalformedPolicyDocumentException: An error occurred (MalformedPolicyDocument) when calling the AssumeRole operation: The policy is not in the valid JSON format.
fatal: [localhost]: FAILED! => {"changed": false, "error": {"code": "MalformedPolicyDocument", "message": "The policy is not in the valid JSON format.", "type": "Sender"}, "msg": "An error occurred (MalformedPolicyDocument) when calling the AssumeRole operation: The policy is not in the valid JSON format.", "response_metadata": {"http_headers": {"connection": "close", "content-length": "298", "content-type": "text/xml", "date": "Thu, 09 Nov 2017 12:08:27 GMT", "x-amzn-requestid": "b1ada81d-c546-11e7-972e-17e2cfe93cee"}, "http_status_code": 400, "request_id": "b1ada81d-c546-11e7-972e-17e2cfe93cee", "retry_attempts": 0}}
...ignoring

TASK [sts_assume_role : assert assume role with invalid policy] ****************
task path: /root/ansible/test/integration/targets/sts_assume_role/tasks/main.yml:131
fatal: [localhost]: FAILED! => {
    "assertion": "'Not authorized to perform sts:AssumeRole' in result.msg", 
    "changed": false, 
    "evaluated_to": false
}
```

Reply:
Cannot reproduce this. The tested behaviour seems more likely than the error I saw.